Update Setup Wizard

jonboiser opened the pull request:
### Summary

Update setup wizard to six-part workflow and new settings
![screen shot 2018-08-06 at 5 07 03 pm](https://user-images.githubusercontent.com/10248067/43746862-917f02dc-999b-11e8-9617-9567f859ac04.png)
![screen shot 2018-08-06 at 5 06 48 pm](https://user-images.githubusercontent.com/10248067/43746863-91958868-999b-11e8-87d9-05af9bad03b5.png)
![screen shot 2018-08-06 at 5 06 38 pm](https://user-images.githubusercontent.com/10248067/43746864-91ab3f50-999b-11e8-8e15-9ad6de3c3c60.png)
![screen shot 2018-08-06 at 5 06 17 pm](https://user-images.githubusercontent.com/10248067/43746865-91c21c84-999b-11e8-937a-54e71a705d6a.png)
![screen shot 2018-08-06 at 5 06 01 pm](https://user-images.githubusercontent.com/10248067/43746866-91da025e-999b-11e8-9299-49127c549de3.png)
![screen shot 2018-08-06 at 5 04 05 pm](https://user-images.githubusercontent.com/10248067/43746867-91f3a538-999b-11e8-9dcc-206a1a3cd21e.png)


### Reviewer guidance

1. Compare visual design and copy with [prototype](https://www.figma.com/file/CiaGXfI6OMOjBF9e8hr1c1Ap/Onboarding-0.11-updates?node-id=111%3A74)
1. Check defaults for steps 3-5 with [these user stories](https://docs.google.com/document/d/1U_8bDJf7STIQXhYtjaUyZmlp7_06ps-9Y_2Ez-Wh7jY/edit)
1. Go forward through as many combinations of settings as possible check to see that post-provisioned app is configured properly.
1. Go backward and forward through setup wizard to make sure info is maintained
1. Make sure keyboard-only control works.

### References
<!--
 * references to related issues and PRs
 * links to mockups or specs for new features
 * links to the diffs for any dependency updates, e.g. in iceqube or the perseus plugin
-->

Fixes #4023
…

----

### Contributor Checklist

- [x] Contributor has fully tested the PR manually
- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [x] If there are any front-end changes, before/after screenshots are included
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label

### Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] Reviewer has fully tested the PR manually
- [ ] PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- [ ] External dependencies files were updated (`yarn` and `pip`)
- [ ] Documentation is updated
- [ ] Link to diff of internal dependency change is included
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] Contributor is in AUTHORS.rst

jonboiser committed:
Add test for current setup wizard workflow

jonboiser committed:
Replace Step 2 of setup wizard with Facility PermissionsForm

jonboiser committed:
Update copy in FacilityPermissionsForm

jonboiser committed:
Remove modal explaining different presets

jonboiser committed:
Move FacilityNameForm into FacilityPermissionsForm

jonboiser committed:
Add Guest Access Form

jonboiser committed:
Add require password + create learner account forms

jonboiser committed:
Setup state for new settings

jonboiser committed:
Update copy in superusercredentialsform

jonboiser committed:
Add test for GuestAccessForm

jonboiser committed:
Add test for CreateLearnerAccountForm

jonboiser committed:
Add test for RequirePasswordForLearnersForm

jonboiser committed:
Retain choices made in setup wizard

jonboiser committed:
Fix focusing of textbox

jonboiser committed:
Add test for FacilityPermissionsForm

jonboiser committed:
Give default facility name for personal preset

jonboiser committed:
Update device provisioning logic

jonboiser committed:
Combine can edit name/username options with can signup setting

jonboiser committed:
Add tooltip to Step 3 No Option

jonboiser committed:
Add test for SuperuserCredentialsForm

jonboiser committed:
update FacilityPermissionForm copy

jonboiser committed:
Fix tests

jonboiser committed:
Remove non-existent SET_CURRENT_ATTEMPT mutation

jonboiser committed:
Fix bad .set call

mrpau-richard reviewed (approved):
I tested this PR @jonboiser. All listed in the Reviewer guidance work with this PR.

DXCanas commented on kolibri/plugins/learn/assets/src/modules/examViewer/actions.js:
```diff
@@ -56,11 +55,9 @@ export function setAndSaveCurrentExamAttemptLog(
           calcQuestionsAnswered(store.rootState.examAttemptLogs)
         );
         // Add this attempt log to the Collection for future caching.
-        const examAttemptLogCollection = ExamAttemptLogResource.fetchCollection({
```
Was this a part of the resource refactor? Is this currently broken?

jonboiser replied:
I might have copied the code over wrong. Basically, you can't call `.set` on the return of `fetchCollection`.

DXCanas replied:
Wait so does this need any updates?

DXCanas commented on kolibri/plugins/setup_wizard/assets/src/modules/pluginModule.js:
```diff
@@ -6,24 +6,39 @@ import { permissionPresets } from '../constants';
       facility: {
         name: '',
       },
+      preset: findKey(permissionPresets, preset => preset.default) || '',
```
Any reason to use `findkey` over native `someObject[somekey]`?

DXCanas commented on kolibri/plugins/setup_wizard/assets/src/views/onboarding-forms/FacilityNameForm.vue:
```diff
@@ -1,11 +1,6 @@
-    :submitText="submitText"
-    @submit="setFacilityName"
-  >
+  <div>
```
Might be best to stop calling this component a "form"

jonboiser replied:
Maybe `FacilityNameField`?

DXCanas commented on kolibri/plugins/setup_wizard/assets/src/views/onboarding-forms/SuperuserCredentialsForm.vue:
```diff
@@ -149,29 +158,31 @@
     },
+    beforeDestroy() {
+      // saves data if going backwards in wizard
+      this.saveSuperuserCredentials(false);
```
Hmm. Works well, but double submission on "finish". Not sure that's a problem at all, but having a "submit" call might be redundant. 

Not exactly semantic, but what do you think of using submit _only_ as a "next step" and use `onDestroy` for all the forms? Would retain data across every step - that's not gherk'ed out behavior, but might be nice?

DXCanas reviewed (commented):
- [ ] "More Languages" button not working.


More appropriate for a later bug:
- Keyboard highlight on "back" button at top of form is off

DXCanas reviewed (changes requested):
Missed this on last review ^

jonboiser committed:
Rename FacilityNameForm -> FacilityNameTextbox

jonboiser commented:
@DXCanas I think the change language button is broken because all of the locale files are outdated after renaming things (see #4152 ).

Not sure how to fix the keyboard focus thing. The only CSS rules I see for it are in CoreMenuOption.vue.

1. Renamed FacilityNameForm -> FacilityNameTextbox
1. Simplified the initialization of the the `preset` field in pluginModule

jonboiser committed:
Make the non-formal preset the default
